docs/reprocess-report.md

```diff
@@ -0,0 +1,122 @@
+- insert those `msm_no` together with matching `fingerprint_no` into `http_request_fp` table as if those were actually ingested by `centrifugation.py`
+- update `anomaly` and `confirmed` flags in `measurement` table for the affected measurement according to the logic codified in `calc_measurement_flags()`
+- update `centrifugation.py`: 1) set up-to-date `fingerprint` table checksum in `HttpRequestFPFeeder.__init__()`, 2) bump global `CODE_VER` and `HttpRequestFPFeeder.min_compat_code_ver` (and only it, to avoid rewriting other tables)
+- roll out `openobservatory/pipeline-shovel` and unpause data ingestion
+- reprocess all the previous buckets under GNU Make control
+
+It's possible to try to use `body_simhash` to reprocess _likely-affected_ buckets first to reduce time-to-publication latency, but that's out of the scope of the document.
+
+## Case: new feature-based fingerprint
+
+The goal of special handling of feature-based case is that the case does not depend on voluminous HTTP bodies. So the flags for the dataset can be updated within couple of hours given quite modest computing resources (4 vCPU, 16 GiB RAM, HDD) compared to ≈46 CPU-days needed ingest whole dataset from scratch.
+
+Examples are `Location` redirects and DNS-based redirects to blockpage servers.
+
+E.g. aforementioned [homeline.kg ISP](https://explorer.ooni.torproject.org/measurement/20180126T000430Z_AS8449_pk15Mr2LgOhNOk9NfI2EarhUAM64DZ3R85nh4Z3q2m56hflUGh?input=http:%2F%2Farchive.org) actually serves redirect for a blocked http URI with no `Date` and no `Server` headers that clearly looks like injected HTTP redirect.
+
+This case is almost the same one as the case of a static blockpage: the MetaDB has all the data to follow fast-path updating measurement metadata (`http_request_fp` table, `confirmed` and `anomaly` flags, etc.) with direct DB queries. The downside of fast-path is that it'll lead to duplication of logic between the queries and `centrifugation.py` that may (by mistake) lead to inconsistencies if the logic is not perfectly equivalent.
+
+Overall steps needed to mark existing & future measurements are the same as for HTML blockpage fingerprint with small alterations:
+
+- ...
+- create a temporary table having `msm_no` of the measurements matching the fingerprint _perfectly_ according to features existing in database (e.g. `select msm_no from http_request where headers->>'Location' = 'http://homeline.kg/access/blockpage.html'`, keep in mind that keys of headers are case-sensitive)
+- ...
+- update `centrifugation.py`: set up-to-date `fingerprint` table checksum in `HttpRequestFPFeeder.__init__()`. There is no need to bump `CODE_VER` for feature-based fingerprints as we are 100% confident that reprocessing is not needed and ongoing data processing is paused.
+- ...
```

Comment:
What is this extra step here?

Reply:
As it's said in the paragraph before the list:

> Overall steps needed to mark existing & future measurements are the same as for HTML blockpage fingerprint

So it's  _"roll out `openobservatory/pipeline-shovel` and unpause data ingestion"_.

Can you suggest a better way to represent the same idea without distracting the reader with repetition?

Reply:
If that is what you mean then I suggest you repeat the sentence. It's better to have some repetition, but not be ambiguous.